pkg/reconciler/events/pubsub/pubsub.go

```diff
@@ -75,37 +63,38 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, pubsub *v1alpha1.CloudPu
 	ctx = logging.WithLogger(ctx, r.Logger.With(zap.Any("pubsub", pubsub)))
 
 	pubsub.Status.InitializeConditions()
-	pubsub.Status.ObservedGeneration = pubsub.Generation
 
 	if pubsub.DeletionTimestamp != nil {
 		// No finalizer needed, the pullsubscription will be garbage collected.
 		return nil
 	}
 
-	ps, err := r.reconcilePullSubscription(ctx, pubsub)
-	if err != nil {
-		pubsub.Status.MarkPullSubscriptionFailed("PullSubscriptionReconcileFailed", "Failed to reconcile PullSubscription: %s", err.Error())
-		return err
+	ps, event := r.reconcilePullSubscription(ctx, pubsub)
+	if event != nil {
+		pubsub.Status.MarkPullSubscriptionFailed(reconciledFailedReason, "Failed to reconcile PullSubscription: %s", event.Error())
+		return event
 	}
+	// Set the ObservedGeneration after successful reconcile
+	pubsub.Status.ObservedGeneration = pubsub.Generation
 	pubsub.Status.PropagatePullSubscriptionStatus(&ps.Status)
 
 	// Sink has been resolved from the underlying PullSubscription, set it here.
 	sinkURI, err := apis.ParseURL(ps.Status.SinkURI)
 	if err != nil {
 		pubsub.Status.SinkURI = nil
-		return err
+		return pkgreconciler.NewEvent(corev1.EventTypeWarning, reconciledFailedReason, "Getting sink URI failed with: %s", err)
 	} else {
 		pubsub.Status.SinkURI = sinkURI
 	}
-	return newReconciledNormal(pubsub.Namespace, pubsub.Name)
+	return pkgreconciler.NewEvent(corev1.EventTypeNormal, reconciledSuccessReason, `PullSubscription reconciled: "%s/%s"`, pubsub.Namespace, pubsub.Name)
```

Comment:
nit: instead of `PullSubscription reconciled`, can we make it `CloudPubSubSource reconciled` ?

Reply:
done